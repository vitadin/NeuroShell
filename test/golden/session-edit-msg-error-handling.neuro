%% Test session-edit-msg error handling and validation
\set[_echo_command="true"]

%% Create session with limited messages for bounds testing
\session-new[system="Test assistant for error handling"] error_test
\session-add-usermsg Only message in session

%% Test missing idx parameter
\try \session-edit-msg New content without idx
\get[@error]

%% Test empty content
\try \session-edit-msg[idx=1] 
\get[@error]

%% Test whitespace-only content
\try \session-edit-msg[idx=1]    
\get[@error]

%% Test invalid idx formats
\try \session-edit-msg[idx=abc] Invalid index format
\get[@error]

\try \session-edit-msg[idx=.] Empty dot format
\get[@error]

\try \session-edit-msg[idx=.abc] Invalid dot format
\get[@error]

%% Test out of bounds - reverse order
\try \session-edit-msg[idx=2] Out of bounds reverse
\get[@error]

\try \session-edit-msg[idx=0] Zero index reverse
\get[@error]

%% Test out of bounds - normal order
\try \session-edit-msg[idx=.2] Out of bounds normal
\get[@error]

\try \session-edit-msg[idx=.0] Zero index normal
\get[@error]

%% Test with nonexistent session
\try \session-edit-msg[session=nonexistent, idx=1] Edit nonexistent
\get[@error]

%% Test with empty session
\session-new[system="Empty session"] empty_test
\try \session-edit-msg[session=empty_test, idx=1] Edit empty session
\get[@error]

%% Clean up
\session-delete error_test
\session-delete empty_test