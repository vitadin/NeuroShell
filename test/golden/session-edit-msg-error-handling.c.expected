Setting _echo_command = true
%%> "\\session-new[system=\"Test assistant for error handling\"] error_test"
Created session 'error_test' (ID: 00000001)
%%> "\\silent \\session-activate[id=true] 00000001-0000-4000-8000-000000000001"
%%> "\\session-add-usermsg Only message in session"
Added user message to session 'error_test'
%%> "\\silent \\session-activate[id=true] 00000001-0000-4000-8000-000000000001"
%%> "\\try \\session-edit-msg New content without idx"
%%> "\\session-edit-msg New content without idx"
%%> "\\get[@error]"
@error = idx parameter is required. Usage: \session-edit-msg[idx=N, session=session_id] new_message_content                
\session-edit-msg[idx=.N] new_message_content                                                                              
                                                                                                                           
Examples:                                                                                                                  
  \session-edit-msg[idx=1] Edit the last message content                         %% Reverse order: 1=last, 2=second-to-last
  \session-edit-msg[idx=2] Edit second-to-last message                          %% Reverse order                           
  \session-edit-msg[idx=.1] Edit the first message content                      %% Normal order: .1=first, .2=second       
  \session-edit-msg[idx=.3] Edit the third message                              %% Normal order                            
  \session-edit-msg[session=work, idx=1] Edit last message in work session     %% Specific session                         
  \session-edit-msg[idx=1] ${new_content}                                       %% Using variable                          
                                                                                                                           
Options:                                                                                                                   
  idx     - Message index (required): N for reverse order (1=last), .N for normal order (.1=first)                         
  session - Session name or ID (optional, defaults to active session)                                                      
                                                                                                                           
Input: New message content to replace the existing content                                                                 
                                                                                                                           
Note: Indexing systems:                                                                                                    
      - Reverse order (idx=N): 1=last message, 2=second-to-last, 3=third-to-last, etc.                                     
      - Normal order (idx=.N): .1=first message, .2=second message, .3=third message, etc.                                 
      Message metadata (ID, role, timestamp) is preserved, only content is changed.                                        
%%> "\\try \\session-edit-msg[idx=1]"
%%> "\\session-edit-msg[idx=1]"
%%> "\\get[@error]"
@error = new message content is required. Usage: \session-edit-msg[idx=N, session=session_id] new_message_content          
\session-edit-msg[idx=.N] new_message_content                                                                              
                                                                                                                           
Examples:                                                                                                                  
  \session-edit-msg[idx=1] Edit the last message content                         %% Reverse order: 1=last, 2=second-to-last
  \session-edit-msg[idx=2] Edit second-to-last message                          %% Reverse order                           
  \session-edit-msg[idx=.1] Edit the first message content                      %% Normal order: .1=first, .2=second       
  \session-edit-msg[idx=.3] Edit the third message                              %% Normal order                            
  \session-edit-msg[session=work, idx=1] Edit last message in work session     %% Specific session                         
  \session-edit-msg[idx=1] ${new_content}                                       %% Using variable                          
                                                                                                                           
Options:                                                                                                                   
  idx     - Message index (required): N for reverse order (1=last), .N for normal order (.1=first)                         
  session - Session name or ID (optional, defaults to active session)                                                      
                                                                                                                           
Input: New message content to replace the existing content                                                                 
                                                                                                                           
Note: Indexing systems:                                                                                                    
      - Reverse order (idx=N): 1=last message, 2=second-to-last, 3=third-to-last, etc.                                     
      - Normal order (idx=.N): .1=first message, .2=second message, .3=third message, etc.                                 
      Message metadata (ID, role, timestamp) is preserved, only content is changed.                                        
%%> "\\try \\session-edit-msg[idx=1]"
%%> "\\session-edit-msg[idx=1]"
%%> "\\get[@error]"
@error = new message content is required. Usage: \session-edit-msg[idx=N, session=session_id] new_message_content          
\session-edit-msg[idx=.N] new_message_content                                                                              
                                                                                                                           
Examples:                                                                                                                  
  \session-edit-msg[idx=1] Edit the last message content                         %% Reverse order: 1=last, 2=second-to-last
  \session-edit-msg[idx=2] Edit second-to-last message                          %% Reverse order                           
  \session-edit-msg[idx=.1] Edit the first message content                      %% Normal order: .1=first, .2=second       
  \session-edit-msg[idx=.3] Edit the third message                              %% Normal order                            
  \session-edit-msg[session=work, idx=1] Edit last message in work session     %% Specific session                         
  \session-edit-msg[idx=1] ${new_content}                                       %% Using variable                          
                                                                                                                           
Options:                                                                                                                   
  idx     - Message index (required): N for reverse order (1=last), .N for normal order (.1=first)                         
  session - Session name or ID (optional, defaults to active session)                                                      
                                                                                                                           
Input: New message content to replace the existing content                                                                 
                                                                                                                           
Note: Indexing systems:                                                                                                    
      - Reverse order (idx=N): 1=last message, 2=second-to-last, 3=third-to-last, etc.                                     
      - Normal order (idx=.N): .1=first message, .2=second message, .3=third message, etc.                                 
      Message metadata (ID, role, timestamp) is preserved, only content is changed.                                        
%%> "\\try \\session-edit-msg[idx=abc] Invalid index format"
%%> "\\session-edit-msg[idx=abc] Invalid index format"
%%> "\\get[@error]"
@error = invalid index 'abc': invalid reverse order index number: strconv.Atoi: parsing "abc": invalid syntax. Usage: \session-edit-msg[idx=N, session=session_id] new_message_content
\session-edit-msg[idx=.N] new_message_content                                                                                                                                         
                                                                                                                                                                                      
Examples:                                                                                                                                                                             
  \session-edit-msg[idx=1] Edit the last message content                         %% Reverse order: 1=last, 2=second-to-last                                                           
  \session-edit-msg[idx=2] Edit second-to-last message                          %% Reverse order                                                                                      
  \session-edit-msg[idx=.1] Edit the first message content                      %% Normal order: .1=first, .2=second                                                                  
  \session-edit-msg[idx=.3] Edit the third message                              %% Normal order                                                                                       
  \session-edit-msg[session=work, idx=1] Edit last message in work session     %% Specific session                                                                                    
  \session-edit-msg[idx=1] ${new_content}                                       %% Using variable                                                                                     
                                                                                                                                                                                      
Options:                                                                                                                                                                              
  idx     - Message index (required): N for reverse order (1=last), .N for normal order (.1=first)                                                                                    
  session - Session name or ID (optional, defaults to active session)                                                                                                                 
                                                                                                                                                                                      
Input: New message content to replace the existing content                                                                                                                            
                                                                                                                                                                                      
Note: Indexing systems:                                                                                                                                                               
      - Reverse order (idx=N): 1=last message, 2=second-to-last, 3=third-to-last, etc.                                                                                                
      - Normal order (idx=.N): .1=first message, .2=second message, .3=third message, etc.                                                                                            
      Message metadata (ID, role, timestamp) is preserved, only content is changed.                                                                                                   
%%> "\\try \\session-edit-msg[idx=.] Empty dot format"
%%> "\\session-edit-msg[idx=.] Empty dot format"
%%> "\\get[@error]"
@error = invalid index '.': invalid normal order index format (use .1, .2, .3, etc.). Usage: \session-edit-msg[idx=N, session=session_id] new_message_content
\session-edit-msg[idx=.N] new_message_content                                                                                                                
                                                                                                                                                             
Examples:                                                                                                                                                    
  \session-edit-msg[idx=1] Edit the last message content                         %% Reverse order: 1=last, 2=second-to-last                                  
  \session-edit-msg[idx=2] Edit second-to-last message                          %% Reverse order                                                             
  \session-edit-msg[idx=.1] Edit the first message content                      %% Normal order: .1=first, .2=second                                         
  \session-edit-msg[idx=.3] Edit the third message                              %% Normal order                                                              
  \session-edit-msg[session=work, idx=1] Edit last message in work session     %% Specific session                                                           
  \session-edit-msg[idx=1] ${new_content}                                       %% Using variable                                                            
                                                                                                                                                             
Options:                                                                                                                                                     
  idx     - Message index (required): N for reverse order (1=last), .N for normal order (.1=first)                                                           
  session - Session name or ID (optional, defaults to active session)                                                                                        
                                                                                                                                                             
Input: New message content to replace the existing content                                                                                                   
                                                                                                                                                             
Note: Indexing systems:                                                                                                                                      
      - Reverse order (idx=N): 1=last message, 2=second-to-last, 3=third-to-last, etc.                                                                       
      - Normal order (idx=.N): .1=first message, .2=second message, .3=third message, etc.                                                                   
      Message metadata (ID, role, timestamp) is preserved, only content is changed.                                                                          
%%> "\\try \\session-edit-msg[idx=.abc] Invalid dot format"
%%> "\\session-edit-msg[idx=.abc] Invalid dot format"
%%> "\\get[@error]"
@error = invalid index '.abc': invalid normal order index number: strconv.Atoi: parsing "abc": invalid syntax. Usage: \session-edit-msg[idx=N, session=session_id] new_message_content
\session-edit-msg[idx=.N] new_message_content                                                                                                                                         
                                                                                                                                                                                      
Examples:                                                                                                                                                                             
  \session-edit-msg[idx=1] Edit the last message content                         %% Reverse order: 1=last, 2=second-to-last                                                           
  \session-edit-msg[idx=2] Edit second-to-last message                          %% Reverse order                                                                                      
  \session-edit-msg[idx=.1] Edit the first message content                      %% Normal order: .1=first, .2=second                                                                  
  \session-edit-msg[idx=.3] Edit the third message                              %% Normal order                                                                                       
  \session-edit-msg[session=work, idx=1] Edit last message in work session     %% Specific session                                                                                    
  \session-edit-msg[idx=1] ${new_content}                                       %% Using variable                                                                                     
                                                                                                                                                                                      
Options:                                                                                                                                                                              
  idx     - Message index (required): N for reverse order (1=last), .N for normal order (.1=first)                                                                                    
  session - Session name or ID (optional, defaults to active session)                                                                                                                 
                                                                                                                                                                                      
Input: New message content to replace the existing content                                                                                                                            
                                                                                                                                                                                      
Note: Indexing systems:                                                                                                                                                               
      - Reverse order (idx=N): 1=last message, 2=second-to-last, 3=third-to-last, etc.                                                                                                
      - Normal order (idx=.N): .1=first message, .2=second message, .3=third message, etc.                                                                                            
      Message metadata (ID, role, timestamp) is preserved, only content is changed.                                                                                                   
%%> "\\try \\session-edit-msg[idx=2] Out of bounds reverse"
%%> "\\session-edit-msg[idx=2] Out of bounds reverse"
%%> "\\get[@error]"
@error = invalid index '2': reverse order index 2 is out of bounds (session has 1 messages). Usage: \session-edit-msg[idx=N, session=session_id] new_message_content
\session-edit-msg[idx=.N] new_message_content                                                                                                                       
                                                                                                                                                                    
Examples:                                                                                                                                                           
  \session-edit-msg[idx=1] Edit the last message content                         %% Reverse order: 1=last, 2=second-to-last                                         
  \session-edit-msg[idx=2] Edit second-to-last message                          %% Reverse order                                                                    
  \session-edit-msg[idx=.1] Edit the first message content                      %% Normal order: .1=first, .2=second                                                
  \session-edit-msg[idx=.3] Edit the third message                              %% Normal order                                                                     
  \session-edit-msg[session=work, idx=1] Edit last message in work session     %% Specific session                                                                  
  \session-edit-msg[idx=1] ${new_content}                                       %% Using variable                                                                   
                                                                                                                                                                    
Options:                                                                                                                                                            
  idx     - Message index (required): N for reverse order (1=last), .N for normal order (.1=first)                                                                  
  session - Session name or ID (optional, defaults to active session)                                                                                               
                                                                                                                                                                    
Input: New message content to replace the existing content                                                                                                          
                                                                                                                                                                    
Note: Indexing systems:                                                                                                                                             
      - Reverse order (idx=N): 1=last message, 2=second-to-last, 3=third-to-last, etc.                                                                              
      - Normal order (idx=.N): .1=first message, .2=second message, .3=third message, etc.                                                                          
      Message metadata (ID, role, timestamp) is preserved, only content is changed.                                                                                 
%%> "\\try \\session-edit-msg[idx=0] Zero index reverse"
%%> "\\session-edit-msg[idx=0] Zero index reverse"
%%> "\\get[@error]"
@error = invalid index '0': reverse order index 0 is out of bounds (session has 1 messages). Usage: \session-edit-msg[idx=N, session=session_id] new_message_content
\session-edit-msg[idx=.N] new_message_content                                                                                                                       
                                                                                                                                                                    
Examples:                                                                                                                                                           
  \session-edit-msg[idx=1] Edit the last message content                         %% Reverse order: 1=last, 2=second-to-last                                         
  \session-edit-msg[idx=2] Edit second-to-last message                          %% Reverse order                                                                    
  \session-edit-msg[idx=.1] Edit the first message content                      %% Normal order: .1=first, .2=second                                                
  \session-edit-msg[idx=.3] Edit the third message                              %% Normal order                                                                     
  \session-edit-msg[session=work, idx=1] Edit last message in work session     %% Specific session                                                                  
  \session-edit-msg[idx=1] ${new_content}                                       %% Using variable                                                                   
                                                                                                                                                                    
Options:                                                                                                                                                            
  idx     - Message index (required): N for reverse order (1=last), .N for normal order (.1=first)                                                                  
  session - Session name or ID (optional, defaults to active session)                                                                                               
                                                                                                                                                                    
Input: New message content to replace the existing content                                                                                                          
                                                                                                                                                                    
Note: Indexing systems:                                                                                                                                             
      - Reverse order (idx=N): 1=last message, 2=second-to-last, 3=third-to-last, etc.                                                                              
      - Normal order (idx=.N): .1=first message, .2=second message, .3=third message, etc.                                                                          
      Message metadata (ID, role, timestamp) is preserved, only content is changed.                                                                                 
%%> "\\try \\session-edit-msg[idx=.2] Out of bounds normal"
%%> "\\session-edit-msg[idx=.2] Out of bounds normal"
%%> "\\get[@error]"
@error = invalid index '.2': normal order index 2 is out of bounds (session has 1 messages). Usage: \session-edit-msg[idx=N, session=session_id] new_message_content
\session-edit-msg[idx=.N] new_message_content                                                                                                                       
                                                                                                                                                                    
Examples:                                                                                                                                                           
  \session-edit-msg[idx=1] Edit the last message content                         %% Reverse order: 1=last, 2=second-to-last                                         
  \session-edit-msg[idx=2] Edit second-to-last message                          %% Reverse order                                                                    
  \session-edit-msg[idx=.1] Edit the first message content                      %% Normal order: .1=first, .2=second                                                
  \session-edit-msg[idx=.3] Edit the third message                              %% Normal order                                                                     
  \session-edit-msg[session=work, idx=1] Edit last message in work session     %% Specific session                                                                  
  \session-edit-msg[idx=1] ${new_content}                                       %% Using variable                                                                   
                                                                                                                                                                    
Options:                                                                                                                                                            
  idx     - Message index (required): N for reverse order (1=last), .N for normal order (.1=first)                                                                  
  session - Session name or ID (optional, defaults to active session)                                                                                               
                                                                                                                                                                    
Input: New message content to replace the existing content                                                                                                          
                                                                                                                                                                    
Note: Indexing systems:                                                                                                                                             
      - Reverse order (idx=N): 1=last message, 2=second-to-last, 3=third-to-last, etc.                                                                              
      - Normal order (idx=.N): .1=first message, .2=second message, .3=third message, etc.                                                                          
      Message metadata (ID, role, timestamp) is preserved, only content is changed.                                                                                 
%%> "\\try \\session-edit-msg[idx=.0] Zero index normal"
%%> "\\session-edit-msg[idx=.0] Zero index normal"
%%> "\\get[@error]"
@error = invalid index '.0': normal order index 0 is out of bounds (session has 1 messages). Usage: \session-edit-msg[idx=N, session=session_id] new_message_content
\session-edit-msg[idx=.N] new_message_content                                                                                                                       
                                                                                                                                                                    
Examples:                                                                                                                                                           
  \session-edit-msg[idx=1] Edit the last message content                         %% Reverse order: 1=last, 2=second-to-last                                         
  \session-edit-msg[idx=2] Edit second-to-last message                          %% Reverse order                                                                    
  \session-edit-msg[idx=.1] Edit the first message content                      %% Normal order: .1=first, .2=second                                                
  \session-edit-msg[idx=.3] Edit the third message                              %% Normal order                                                                     
  \session-edit-msg[session=work, idx=1] Edit last message in work session     %% Specific session                                                                  
  \session-edit-msg[idx=1] ${new_content}                                       %% Using variable                                                                   
                                                                                                                                                                    
Options:                                                                                                                                                            
  idx     - Message index (required): N for reverse order (1=last), .N for normal order (.1=first)                                                                  
  session - Session name or ID (optional, defaults to active session)                                                                                               
                                                                                                                                                                    
Input: New message content to replace the existing content                                                                                                          
                                                                                                                                                                    
Note: Indexing systems:                                                                                                                                             
      - Reverse order (idx=N): 1=last message, 2=second-to-last, 3=third-to-last, etc.                                                                              
      - Normal order (idx=.N): .1=first message, .2=second message, .3=third message, etc.                                                                          
      Message metadata (ID, role, timestamp) is preserved, only content is changed.                                                                                 
%%> "\\try \\session-edit-msg[session=nonexistent, idx=1] Edit nonexistent"
%%> "\\session-edit-msg[session=nonexistent, idx=1] Edit nonexistent"
%%> "\\get[@error]"
@error = failed to find session 'nonexistent': session 'nonexistent' not found (tried both name and ID)
%%> "\\session-new[system=\"Empty session\"] empty_test"
Created session 'empty_test' (ID: 00000003)
%%> "\\silent \\session-activate[id=true] 00000003-0000-4000-8000-000000000003"
%%> "\\try \\session-edit-msg[session=empty_test, idx=1] Edit empty session"
%%> "\\session-edit-msg[session=empty_test, idx=1] Edit empty session"
%%> "\\get[@error]"
@error = session 'empty_test' has no messages to edit
%%> "\\session-delete error_test"
Deleted session 'error_test' (ID: 00000001)
%%> "\\silent \\session-activate"
%%> "\\session-delete empty_test"
Deleted session 'empty_test' (ID: 00000003)
%%> "\\silent \\session-activate"